lib/std/math.zig

```diff
@@ -1279,6 +1279,35 @@ test "lossyCast" {
     try testing.expect(lossyCast(u32, @as(f32, maxInt(u32))) == maxInt(u32));
 }
 
+/// Performs linear interpolation between *a* and *b* based on *t*.
+/// *t* must be in range 0.0 to 1.0. T must be a float type.
+///
+/// This does not guarantee returning *b* if *t* is 1 due to floating-point errors.
+/// This is monotonic.
+pub fn lerp(a: anytype, b: anytype, t: anytype) @TypeOf(a, b, t) {
+    const T = @TypeOf(a, b, t);
+
+    comptime if (!std.meta.trait.isFloat(T))
+        @compileError("T must be a float type");
+
+    assert(t >= 0 and t <= 1);
+    return @mulAdd(T, b - a, t, a);
+}
+
+test "lerp" {
+    try testing.expectEqual(@as(f64, 75), lerp(50, 100, 0.5));
+    try testing.expectEqual(@as(f32, 43.75), lerp(50, 25, 0.25));
+    try testing.expectEqual(@as(f64, -31.25), lerp(-50, 25, 0.25));
+
+    try testing.expectApproxEqRel(@as(f32, -7.16067345e+03), lerp(-10000.12345, -5000.12345, 0.56789), 1e-19);
+    try testing.expectApproxEqRel(@as(f64, 7.010987590521e+62), lerp(0.123456789e-64, 0.123456789e64, 0.56789), 1e-33);
+
+    try testing.expectEqual(@as(f32, 0.0), lerp(@as(f32, 1.0e8), 1.0, 1.0));
```

Comment:
This however is something that confuses me about peer type resolution: why can I not write `lerp(1.0e8, 1.0, 1.0)` here just like above? It doesn't pass otherwise.

Seems like it should work because doesn't it work like this? `lerp(1.0e8, 1.0, 1.0)` will be `comptime_float` and then that's automatically casted to `f32` because `expectEqual` is doing `expected: anytype, actual: @TypeOf(expected)`, so why doesn't it work here?

Reply:
It's not the peer type resolution - it's due to the rounding

The fully-precise answer here is `1.0`, which `comptime_float` (`f128` w.r.t. precision) computes exactly. Using `f32` results in a rounded `0.0`

Reply:
I see. Looks like it's fine as-is then.